File: bench/sum_fenv.js
env = {}

function sum(n){
    with(env) {
        var total = 0;
        for (var i = 0; i <= n; i++) {
            total = total + i;
        }
        return total;
    }
}
console.time("cpu time");
for (var i = 0; i < 10000; i++) {
    sum(10000);
}
console.timeEnd("cpu time");

File: bench/sum_fenv.lua

function sum(n)
   env = _ENV
   _ENV = env
   total = 0
   for i=0,n do
      total = total + i
   end
   return total
end

start_time=os.clock()
for i=0,10000 do
   sum(10000)
end
print("cpu time: ".. (os.clock()-start_time)*1000)

